cmd/notary/tuf.go

```diff
@@ -533,7 +533,7 @@ func tokenAuth(trustServerURL string, baseTransport *http.Transport, gun string,
 	}
 
 	ps := passwordStore{anonymous: readOnly}
-	tokenHandler := auth.NewTokenHandler(authTransport, ps, gun, "push", "pull")
```

Comment:
On the one hand, the actions doesn't hide the `push` and `pull`, on the other hand it seems like then all the code would have to know about the exact auth mechanism?  It may make sense for most things to just know whether it requires read access or read and write access.

Possibly if you wanted to translate that to a constant instead (READ or READWRITE), that'd work?  I think we should abstract the fact that read access is sent as "pull" to the auth realm, and write access is sent as "push", away from most of the code, though.